### PR TITLE
Feat:  added product listing page

### PR DESCRIPTION
**Pull request by aryan-13**
Added the following: 
- Products in the DB file
- Route for the Product Listing page
- Components - FilterBar, Product Cards
- Added the method to fetch data and render it on the product page

image

### CODE REVIEW

File: src/Components/FilterBar.jsx [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+        <span className="checkmark"></span>
+      </label>
+      <label className="check-list">Shirts
+        <input type="checkbox"/>
+        <span className="checkmark"></span>
+      </label>
+      <label className="check-list">Jackets
+        <input type="checkbox"/>
+        <span className="checkmark"></span>
+      </label>
```
Comment (dikshadhote) [inline src/Components/FilterBar.jsx:69]: Put the categories list in array and loop through it as categories can change or you can use caegories DB and fetch from it

File: src/Components/FilterBar.jsx [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+      <div className="slider-container">
+        <input type="range" list="tickmarks" min="10" max="50" step="10" className="slider"/>
+        <datalist id="tickmarks">
+          <option value="10"></option>
+          <option value="20"></option>
+          <option value="30"></option>
+          <option value="40"></option>
+          <option value="50"></option>
+        </datalist>
```
Comment (dikshadhote) [inline src/Components/FilterBar.jsx:87]: Use map for repetitive code

File: src/pages/ProductListing.jsx [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+import Navbar from '../Components/Navbar'
+import axios from 'axios';
+import FilterBar from '../Components/FilterBar'
+import ProductCard from '../Components/ProductCard'
+function ProductListing() {
+  const [products, setProducts] = React.useState([]);
+  const getData = async () => {
+     const data = await axios.get('/api/products');
+    try{
+      console.log(data.data.products);
```
Comment (dikshadhote) [inline src/pages/ProductListing.jsx:12]: Remove console statements before deployment

File: src/pages/ProductListing.jsx [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+      setProducts(data.data.products);
+    }
+    catch{
+      console.log("error")
```
Comment (dikshadhote) [inline src/pages/ProductListing.jsx:16]: Remove console statements before deployment.
You can use throw(error)

**Review by dikshadhote — CHANGES_REQUESTED** [review]
Overall code cleans clean. Please make the request changes. 
The preview is not available check deploy logs and resolve it.
image

**Review by lannisterrr — COMMENTED** [review]
Code seems fine, other than the points mentioned by diksha.

File: src/Components/FilterBar.jsx (hunk 1 above)
Comment (aryan-13) [reply src/Components/FilterBar.jsx:69]: I added the method to fetch categories, but I'll be adding the looping part in the next PR.

File: src/pages/ProductListing.jsx (hunk 4 above)
Comment (aryan-13) [reply src/pages/ProductListing.jsx:16]: Yes, I'll remove them!

**Pull request merged**